include/DDA/DDAVFSolver.h

```diff
@@ -525,11 +525,12 @@ class DDAVFSolver
     virtual inline CPtSet& getCachedPointsTo(const DPIm& dpm)
     {
         if (isTopLevelPtrStmt(dpm.getLoc()))
-            return getCachedTLPointsTo(dpm);
```

Comment:
It seems to me that we can use `getCachedTLPointsTo ` here?

Reply:
`getCachedPointsTo` is used non-`const`ly (to union points-to sets, etc.), and getCachedTLPointsTo in `FlowDDA` relies on `getPts`, so `getCachedTLPointsTo` needs to be `const`.

Another way to overcome this is to add functions into these classes which perform the union and make `getCachedPointsTo` `const`. For example, instead of:
```
    PointsTo &pt = getCachedPointsTo(dpm);
    pt |= someOtherPt;  // Or unionDDAPts(pt, someOtherPt), etc.
```

We would do:
```
    unionDDAPts(dpm, someOtherPt);  // Overload similar to unionPts in PTData
```

The only issue with this is that we would then internally need a non-`const` version of `getCachedPointsTo` for the overload to use. I'd have to study the code closer to think of a cleaner way unless you have some suggestions, I don't think I've really looked at DDA before.

Reply:
In fact, it seems like we *need* a cleaner way, on second thought. I'll think about how to go about fixing it.